src/graph/backend/dnnl/kernels/sdp_decomp_config.cpp

```diff
@@ -41,14 +46,15 @@ bool sdp_decomp_config_t::initial_check(const std::shared_ptr<subgraph_t> &sg,
 
     // Check batch size compatibility.
     dims wei2_user_dims = ltw(inputs[graph_inport[4]]).vdims();
-    if (batch_size != wei1_user_dims[0] || batch_size != wei2_user_dims[0]) {
-        return false;
-    }
+    VCHECK_SDP_DECOMP(
+            batch_size == wei1_user_dims[0] && batch_size == wei2_user_dims[0],
+            false, "Batch size mismatch");
 
     // Check scale size
     if (graph_inport[2] != -1) {
         auto scale_sz = ltw(inputs[graph_inport[2]]).nelems();
-        if (scale_sz != 1) return false;
+        VCHECK_SDP_DECOMP(scale_sz == 1, false,
+                "Scale size in sdp should be 1, but got %d", scale_sz);
```

Comment:
Better to clarify that this is a limitation of the decomp kernel, not sdp.
```suggestion
                "sdp_decomp_kernel_t only supports single scale value, but got %d", scale_sz);
```

Reply:
Okay i see you already have kernel name in the macro, but seems still need to refine. This is a duplication in your example:
```
onednn_verbose,v1,graph,create:check,sdp_decomp,sdp_decomp_kernel_t: initial check failed,src/graph/backend/dnnl/kernels/sdp_decomp.cpp:56
```

Reply:
has removed the duplicated info. thanks for your advice.